circle/functions

```diff
@@ -581,28 +604,120 @@ update_chart_in_repo() {
 
       # auto merge updates to https://github.com/bitnami/charts
       if [[ $REPO_TO_UPDATE == "https://github.com/bitnami/charts" ]]; then
-        info "Auto-merging $CHART_NAME-$CHART_VERSION_FIRST_UPDATE..."
-        git checkout master >/dev/null
-        git merge --no-ff $CHART_NAME-$CHART_VERSION_FIRST_UPDATE >/dev/null
-        git remote remove origin >/dev/null
-        git remote add origin https://$GITHUB_USER@$(echo ${REPO_TO_UPDATE/https:\/\/}).git >/dev/null
-        git push origin master >/dev/null
+        git_automerge_branch $REPO_TO_UPDATE "${CHART_NAME}-${CHART_VERSION_FIRST_UPDATE}"
       fi
     fi
 
     info "Cleaning up old branches..."
-    git fetch development >/dev/null
-    for branch in $(git branch --remote --list development/$CHART_NAME-* | sed 's?.*development/??' | grep -v "^$CHART_NAME-$CHART_VERSION_FIRST_UPDATE$")
-    do
-      log "Deleting $branch..."
-      git push development :$branch >/dev/null
-    done
+    git_cleanup_old_branches "${CHART_NAME}-*" "^$CHART_NAME-$CHART_VERSION_FIRST_UPDATE$"
   else
     warn "Chart release/updates skipped!"
   fi
-  
+
   info "Cleaning up repository"
   # Exiting from charts/ repository
   cd ..
   rm -rf charts
 }
+
+# Expects three parameters: <repo to update> <distro codename> and <docker image digest>
+update_minideb_derived() {
+    if [ -z "$1" ] || [ -z "$2" ] || [ -z "$3" ]
+    then
+        error "Wrong arguments passed to update_minideb_extras function"
+        exit 1
+    fi
+
+    REPO_TO_UPDATE=$1
+    DIST=$2
+    DIST_BUILD_ID=$3
+    BASENAME=bitnami/minideb
+
+    echo "Cloning '$REPO_TO_UPDATE' repo..."
+    if ! git clone --quiet --single-branch $REPO_TO_UPDATE minideb-extras; then
```

Comment:
I would not clone to minideb-extras as you will be using this generic function for `minideb-extras` and `minideb-runtimes`

Reply:
Fixed